fili-core/src/main/java/com/yahoo/bard/webservice/web/DataApiRequest.java

```diff
@@ -290,6 +327,24 @@ public DataApiRequest(
     }
 
     /**
+     * Times the execution of the specified producer, making sure to stop timing in a finally clause.
+     *
+     * @param generator  The producer to be timed
+     * @param timerName  The name to use for the timer
+     * @param <T>  The type of the object produced
+     *
+     * @return The value returned by the generator
+     */
+    private <T> T timeGenerator(Producer<T> generator, String timerName) {
```

Comment:
Fair enough. I dug in a bit deeper too and it seems you are indeed right. That said, we should open an issue to revisit this, since it would help the RequestLog piece immensely to be able to be annotation-driven, rather than code-driven.

That said, we should make sure the timings are internal to the methods that are getting timed, since callers don't really need this complexity, and the method is better suited to handling it. (note: I've not looked at your updates yet, so this may already be the case).